### PR TITLE
feat: adding new /decide endpoint

### PR DESCRIPTION
**Pull request by pawels-optimizely**
## Summary

Adding new endpoint /decide. 
it takes one optional  URL parameter "keys" which can be a single or multiple feature keys and the body:

```
type DecideBody struct {
	UserID         string                 `json:"userId"`
	UserAttributes map[string]interface{} `json:"userAttributes"`
	DecideOptions  []string               `json:"decideOptions"`
}
```
with DecideOptions values: 
```
"DISABLE_DECISION_EVENT",  "ENABLED_FLAGS_ONLY", "IGNORE_USER_PROFILE_SERVICE",
"EXCLUDE_VARIABLES", "INCLUDE_REASONS"
```
When "keys" parameter is not provided then it displays all the available features.

## Prerequisites
Following PR needs to be merged: 
https://github.com/optimizely/fullstack-sdk-compatibility-suite/pull/509

### CODE REVIEW

File: pkg/handlers/decide.go
Comment (msohailhussain) [inline pkg/handlers/decide.go:78]: may be rename it to `decideOuts` or array, more understandable.

File: pkg/handlers/decide.go
Comment (msohailhussain) [inline pkg/handlers/decide.go:81]: make it `decideOut`

**Review by msohailhussain — APPROVED** [review]
lgtm

File: pkg/handlers/decide_test.go
Comment (jaeopt) [inline pkg/handlers/decide_test.go:239]: Aren't we supposed to have "Variables" non-empty here?  If not, can we modify the tests to validate Variables?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good.
I see "decideForKeys" support missing. Is it by design?
Suggested to change a test case.

File: api/openapi-spec/openapi.yaml
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:44]: Please add appropriate response, what exact object keys will be returned.

File: api/openapi-spec/openapi.yaml
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:33]: dont' we have any other param?

File: api/openapi-spec/openapi.yaml
Comment (msohailhussain) [inline api/openapi-spec/openapi.yaml:234]: again DecideContext?

File: api/openapi-spec/openapi.yaml
Comment (oakbani) [inline api/openapi-spec/openapi.yaml:53]: nit. move requestbody before responses

File: api/openapi-spec/openapi.yaml
Comment (oakbani) [inline api/openapi-spec/openapi.yaml:414]: I think you can define all these possible values as enums. so that open spec fails if any other string option is passed within the options

**Review by oakbani — CHANGES_REQUESTED** [review]
Looks good overall. Minor changes requested.

**Review by oakbani — COMMENTED** [review]
open api spec looks good.

File: go.mod
Comment (mjc1283) [inline go.mod:15]: This should not be merged/released until we have a full release (1.5.2) of Go SDK that can be used.

**Review by mjc1283 — CHANGES_REQUESTED** [review]
LGTM, but let's wait until Go SDK release to merge/release this.

**Review by jaeopt — APPROVED** [review]
LGTM with nits - improve test names

**Review by mjc1283 — APPROVED** [review]

**Pull request merged**
